proposals/0000-add-support-for-css-color-function.md

```diff
@@ -0,0 +1,109 @@
+
+## Detailed design
+
+### JS Changes
+
+1. Parse the color() function in [normalizeColor](https://github.com/facebook/react-native/blob/63213712125795ac082597dad2716258b90cdcd5/packages/normalize-color/index.js#L235)
+
+```js
+const COLOR_SPACE = /display-p3|srgb/
+
+function getMatchers() {
+  if (cachedMatchers === undefined) {
+    cachedMatchers = {
+      color: new RegExp(
+        'color(' +
+          call(COLOR_SPACE, NUMBER, NUMBER, NUMBER) +
+          '|' +
+          callWithSlashSeparator(COLOR_SPACE, NUMBER, NUMBER, NUMBER, NUMBER) +
+          ')'
+      ),
+```
+
+2. Include the color space in the return value of [StyleSheet processColor](https://github.com/facebook/react-native/blob/63213712125795ac082597dad2716258b90cdcd5/packages/react-native/Libraries/StyleSheet/processColor.js) and [Animated processColor](https://github.com/facebook/react-native/blob/main/packages/react-native/Libraries/Animated/nodes/AnimatedColor.js)
+
+```js
+return { ["display-p3"]: true, r, g, b, a };
+```
+
+### iOS Changes
+
+1. Update [RCTConvert](https://github.com/facebook/react-native/blob/781b637db4268ad7f5f3910d99ebb5203467840b/packages/react-native/React/Base/RCTConvert.m#L881) to handle setting new color values for color space. If color space is not included preserve existing behavior.
+
+2. Update [RCTConversions](https://github.com/facebook/react-native/blob/16ad818d21773cdf25156642fae83592352ae534/packages/react-native/React/Fabric/RCTConversions.h#L37) to handle setting new color values for color space. If color space is not included preserve existing behavior.
+
+3. Update [RCTTextPrimitivesConversions](https://github.com/facebook/react-native/blob/ac1cdaa71620d5bb4860237cafb108f6aeae9aef/packages/react-native/ReactCommon/react/renderer/textlayoutmanager/platform/ios/react/renderer/textlayoutmanager/RCTTextPrimitivesConversions.h#L116) to handle setting new color values for color space. If color space is not included preserve existing behavior.
+
+```objc
+[UIColor colorWithDisplayP3Red:components.red green:components.green blue:components.blue alpha:components.alpha]
+```
+
+### Android Changes
+
+1. Enable the wide color gamut in [ReactActivity](https://github.com/facebook/react-native/blob/7625a502960e6b107e77542ff0d6f40fbf957322/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/ReactActivity.java#L22) if available.
+
+```java
+Window window = getWindow();
+boolean isScreenWideColorGamut = window.getDecorView().isScreenWideColorGamut()
+if (isScreenWideColorGamut) {
+  window.setColorMode(PixelFormat.COLOR_MODE_WIDE_COLOR_GAMUT)
+}
+```
```

Comment:
If this is a change we apply to all the `ReactActivity`, you have to explain in this RFC why it will not cause harm to the existing code that uses only sRGB values.
Always assume that your readers have no idea how things works.